### ISSUE TITLE
How do I use it given a network with multiple inputs?

### ISSUE THREAD

**Issue by Miail** [maxpumperla/hyperas#81]
Given a Network like this: 

```
def model3():

    #stride = 2
    #dim = 40
    #window_height = 5
    #splits = ((40-5)+1)/2 = 18
    next(test_generator())
    next(train_generator(batch_size))

    kernel_number = [{choice{80,100,120,140,160,180,200}} for i in range(splits)]
    kernel_shape  = [{choice{(window_height,3),(window_height,5),(window_height,8),(window_height,11)}} for i in range(splits)]
    pool_shape = [{choice{(1,3),(1,5),(1,8),(1,11)}} for i in range(splits)]
    list_of_input = [Input(shape = (window_height,total_frames_with_deltas,3)) for i in range(splits)]
    list_of_conv_output = []
    list_of_max_out = []
    for i in range(splits):
        list_of_conv_output.append(Conv2D(filters = kernel_number[i] , kernel_size = kernel_shape[i] , activation = 'relu')(list_of_input[i]))
        list_of_max_out.append((MaxPooling2D(pool_size=((1,11)))(list_of_conv_output[i])))

    merge = keras.layers.concatenate(list_of_max_out)
    print merge.shape
    reshape = Reshape((total_frames/total_frames,-1))(merge)

    dense1 = Dense(units = 1000, activation = 'relu',    name = "dense_1")(reshape)
    dense2 = Dense(units = 1000, activation = 'relu',    name = "dense_2")(dense1)
    dense3 = Dense(units = 145 , activation = 'softmax', name = "dense_3")(dense2)
    #dense4 = Dense(units = 1, activation = 'linear', name = "dense_4")(dense3)


    model = Model(inputs = list_of_input , outputs = dense3)
    model.compile(loss="categorical_crossentropy", optimizer="adam" , metrics = [metrics.categorical_accuracy])
```

How do i set hyperparamter  for kernel_number, kernel_shape and pool shape for each  input?
    `kernel_number = [{choice{80,100,120,140,160,180,200}} for i in range(splits)]`

Doing it like this give me invalid syntax.

**Comment by maxpumperla** [comment]
It seems you're using what we call "intermediate functions", would you mind checking out this example and tell me if it works for you? In principle, there's nothing fundamentally in your example that shouldn't work:

https://github.com/maxpumperla/hyperas/blob/master/examples/use_intermediate_functions.py

**Comment by Chandrahas1991** [comment]
Hi @maxpumperla, does x_train, y_train, x_test, y_test all have to be tensors ? 

If i have multiple input tensors to my network, can i wrap all of them in a dict ? 

Example : 
`return {'Input_1_train' : ip1_train_tensor, 'Input_2_train' : ip2_train_tensor}, { 'Output_train' : op_train_tensor}, {'Input_1_test' : ip1_test_tensor, 'Input_2_test' : ip2_test_tensor}, { 'Output_test' : op_test_tensor} `

**Comment by amw5g** [comment]
I, too, would like to use multiple inputs (single output) to a model.  This is because different parts of my data need different processing inside the model (e.g., batchnormalization or embedding).  So I currently use named inputs like so:
`'X_input': X_train, 'foo_input': foo_train, 'bar_input': bar_train`
In reality, there are more than 3 inputs.  and I have similarly named validation inputs for validating
So before I go about wrapping them all up in a dictionary, and testing/debugging, is there an agreed upon method to pass multiple inputs to hyperopts?

**Comment by amw5g** [comment]
I just submitted a PR that showcases an example of using multiple inputs within a jupyter notebook.  It builds off the simple example.
https://github.com/maxpumperla/hyperas/pull/196
